internal/models/forwarder_v2_otlp.go

```diff
@@ -0,0 +1,119 @@
+package models
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	otlpcommon "go.opentelemetry.io/proto/otlp/common/v1"
+	otlplogs "go.opentelemetry.io/proto/otlp/logs/v1"
+	proto "google.golang.org/protobuf/proto"
+)
+
+type OtlpForwarderConfig struct {
+	Endpoint string            `json:"endpoint,omitempty"`
+	Headers  map[string]string `json:"headers,omitempty"`
+}
+
+type ForwarderOtlpV2 struct {
+	Type   string              `json:"type" enum:"otlp"`
+	Config OtlpForwarderConfig `json:"config"`
+	client *http.Client        `json:"-"`
+}
+
+// call sends a single log record
+func (f *ForwarderOtlpV2) call(ctx context.Context, record *LogRecord) error {
+	// Initialize client if nil
+	if f.client == nil {
+		f.client = &http.Client{}
+	}
+
+	// Convert single log to OTLP protobuf msg format
+	otlpLogs, err := ConvertToOtlpLogs([]*LogRecord{record})
+	if err != nil {
+		return err
+	}
+
+	// Marshal the OTLP logs to protobuf bytes
+	data, err := proto.Marshal(otlpLogs)
+	if err != nil {
+		return fmt.Errorf("failed to marshal OTLP logs: %w", err)
+	}
+
+	// Create HTTP POST request
+	req, err := http.NewRequestWithContext(ctx, "POST", f.Config.Endpoint, bytes.NewBuffer(data))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	// Set Headers
+	req.Header.Set("Content-Type", "application/x-protobuf")
+	for k, v := range f.Config.Headers {
+		req.Header.Set(k, v)
+	}
+
+	// Send the request
+	resp, err := f.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		body, _ := io.ReadAll(resp.Body)
+		errMsg := fmt.Sprintf("unexpected status code: %d, body: %s", resp.StatusCode, string(body))
+		fmt.Println("OTLP forwarder error:", errMsg)
+		return fmt.Errorf(errMsg)
+	}
+
+	return nil
+}
+
+// ConvertToOtlpLogs converts a slice of LogRecord to OTLP LogsData format
+func ConvertToOtlpLogs(records []*LogRecord) (*otlplogs.LogsData, error) {
+	otlpLogs := &otlplogs.LogsData{
+		ResourceLogs: []*otlplogs.ResourceLogs{
+			{
+				ScopeLogs: []*otlplogs.ScopeLogs{
+					{
+						LogRecords: []*otlplogs.LogRecord{},
+					},
+				},
+			},
+		},
+	}
+
+	for _, r := range records {
+		lr := &otlplogs.LogRecord{
+			TimeUnixNano: uint64(r.Timestamp.UnixNano()),
+			Body: &otlpcommon.AnyValue{
+				Value: &otlpcommon.AnyValue_StringValue{
+					StringValue: r.Fields["body"],
```

Comment:
What if the field `body` does not exist in the message?

Can you provide an empty string as a default value?